lightning/src/ln/onion_route_tests.rs

```diff
@@ -412,54 +423,63 @@ fn test_onion_failure() {
 		msg.reason = onion_utils::build_first_hop_failure_packet(&onion_keys[1].shared_secret[..], PERM|NODE|3, &[0;0]);
 	}, ||{
 		nodes[2].node.fail_htlc_backwards(&payment_hash);
-	}, false, Some(PERM|NODE|3), Some(NetworkUpdate::NodeFailure{node_id: route.paths[0][1].pubkey, is_permanent: true}));
+	}, false, Some(PERM|NODE|3), Some(NetworkUpdate::NodeFailure{node_id: route.paths[0][1].pubkey, is_permanent: true}), Some(route.paths[0][1].short_channel_id));
 	let (_, payment_hash, payment_secret) = get_payment_preimage_hash!(nodes[2]);
 
+	// Our immediate peer sent UpdateFailMalformedHTLC because it couldn't understand the onion in
+	// the UpdateAddHTLC that we sent. Thus, short_channel_id is None because there's no sense
```

Comment:
Hmm... @TheBlueMatt for these test cases presumably we don't give a `NetworkUpdate` since it's our immediate peer. But for the case of `BADONION` inside `process_onion_failure`, should we return `NetworkUpdate::ChannelClosed`? The latter case is currently untested as noted earlier.